### PR TITLE
Initial command line interface

### PR DESCRIPTION
**Pull request by zmoon**
I created an initial command line interface (CLI) for MM. As part of this I also started modernizing the setup/build config.

**Ideas**:
* more information in the header?
* treat the control file location as the working directory by default?
* [x] docs (maybe [sphinx-click](https://sphinx-click.readthedocs.io/en/latest/) will work; can hide the module from API docs?)
  - also update the main install page, mentioning the CLI (think I will wait to do this)
* [ ] support auto job submission, initially at least on Hera
  - maybe provide a few templates but also allow custom

**Note**: before merging to main, probably should remove the `.dev` from the `__version__` in the top-level `__init__.py`

### CODE REVIEW

**Comment by zmoon** [comment]
Here's an example of the output, for my idealized example:

image

Let me know any questions/comments/suggestions regarding behavior and features. Looking for people to try it out. @rschwant @dwfncar @Duseong

**Comment by zmoon** [comment]
The one where it succeeded (after I added `typer` to the conda env) was using Sphinx v4.4.0, but then with v4.5.0 it started failing.

**Comment by zmoon** [comment]
Now the CRN URLs in the docs are not working.

**Comment by tiangolo** [comment]
I just released Typer `0.4.1` that should handle the Typer issue. :rocket: :nerd_face:

**Comment by zmoon** [comment]
@rschwant can you check the package metadata in `setup.cfg`? I think the auto-job submission I can try in another PR. I want to get this one in sooner so I can start updating the install docs.

**Review by rschwant — APPROVED** [review]
Looks looks great. I tested it and it works. As a feature for later. It will be good to use this to submit jobs to Hera instead of the scripts I have here: 
/MELODIES-MONET/examples/submit_jobs/run_melodies_monet.py
/MELODIES-MONET/examples/submit_jobs/submit_hera.sh

I added in one comment to change spelling of my name too.

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+name = melodies-monet
+version = attr: melodies_monet.__version__
+description = MELODIES MONET unified verification package
+long_description = file: README.md
+long_description_content_type = text/markdown
+url = https://github.com/NOAA-CSL/MELODIES-MONET
+license = Apache 2.0
+author = Rebecca Schwantes
+author_email = rebecca.schwantes@noaa.gov
+maintainer = Rebecaa Schwantes
```
[the remark this answers is not in the document]
Comment (zmoon) [reply setup.cfg:11]: oops! thanks

**Comment by zmoon** [comment]
Thanks @rschwant

**Comment by zmoon** [comment]
Related to https://github.com/NOAA-CSL/MELODIES-MONET/issues/117

**Pull request merged**